enhancements/E-0001-secret-handling.md

```diff
@@ -0,0 +1,748 @@
+      - This mechanism should have an opt-out hatch if one has good reasons to include specific `SecretSync` objects with hardcoded sources.
+      - This mechanism will probably turn out to be validatable at packaging- AND run-time. Validating at packaging-time would left-shift surfacing of guardrail violations and save a lot of iteration time!
+      - Important: this guardrail cannot be used for defensive security against malicious packagers because it would be easy to put an evil SecretSync object into a wrapper object like an OpenShift `Template` (or another `Package`), as PKO cannot possibly understand and correctly detect and unpack all possible forms of a "trojan horse" object.
+        - personal recommendation from Josh: If you want to be sure about the authenticity of a package image, implement validation of sigstore signatures on package images.
+
+- The PackageManifest API has to be changed so that `Secret` references can be specified and aliased. References specify:
+  - an alias name to be used with newly introduced resolving functions in the package templates
+  - when package is installable cluster-scoped: a namespaced name of the source secret
+  - when package is installable namespace-scoped: a name of the source secret which is expected in the Package's target namespace
+  - ~if the reference is optional or not (`TODO`: clarify implications - probably better off to start without optional secrets and introduce them later when needed. Big point: how to handle optionality without polling for secret's to appear? The package reconciler doesn't declaratively re-template the ObjectDeployment except, when .spec.image changes)~
+
+- The `(Cluster)Package` API has to be changed to
+  - allow the specification of an `ImagePullSecret` for pulling the package image itself from a private registry.
+  - allow an end-user to supply a list of secret references and bind them to the alias names described in the `PackageManifest` by the packaging person.
+
+- The `kubectl-package` cli has to be changed to
+  - include validation for the `secret-sync-fence` guardrail
+  - include ways for an end-user to inspect a package image and get a list of secret references that the `(Cluster)Package` requires.
+    - This could be part of an overall `kubectl package inspect docker://quay.io/foo/bar-package` command that shows interesting information about a package to the end-user.
+
+- for handling `ConfigMaps` the described changes could either be duplicated in all affected APIs with `s/Secret/ConfigMap/` or the described API could be made generic (what are the implications of either? I can think of questions about caching and watching/caching too many different object kinds to stay performant).
```

Comment:
`ConfigMap` usage should be discussed in it's own concept.

Reply:
Removed :)